src/states/node.rs

```diff
@@ -2205,10 +2199,10 @@ impl Node {
                 debug!("{:?} Disconnecting from timed out peer {:?}", self, peer_id);
                 let _ = self.crust_service.disconnect(peer_id);
             }
+            let terminate = self.purge_invalid_rt_entries(outbox);
             self.merge_if_necessary();
-
             outbox.send_event(Event::Tick);
-            return true;
+            return terminate;
```

Comment:
This boolean is backwards. Per the description of this function (`handle_timeout`):

> Returns true if the calling node should keep running, false for terminate or restart.

Whereas in `purge_invalid_rt_entries`, we return `true` if we get a `Transition::Terminate`.

We could fix this issue and avoid any future confusion by returning a `Transition` from this func.